types/graphql/execution/execute.d.ts

```diff
@@ -42,9 +42,9 @@ export interface ExecutionContext {
  *   - `errors` is included when any errors occurred as a non-empty array.
  *   - `data` is the result of a successful execution of the query.
  */
-export interface ExecutionResult {
+export interface ExecutionResult<TData> {
     errors?: ReadonlyArray<GraphQLError>;
-    data?: { [key: string]: any };
+    data?: Readonly<TData>;
```

Comment:
`Readonly` will probably cause some bugs for people attempting to consume + transform the data after execution.

Probably just make it `TData` here instead to reduce the risk of a breaking change.